libcudacxx/include/cuda/std/__complex/inverse_hyperbolic_functions.h

```diff
@@ -39,57 +39,364 @@
 
 _CCCL_BEGIN_NAMESPACE_CUDA_STD
 
+// Specialization of 1/sqrt(x), uses device-only rsqrtf/rsqrt when viable.
+// Sometimes compilers can optimize 1/sqrt(x) so we don't suffer the slowdown of the divide.
+// It can happen that the double 1/sqrt() on device gets optimized better than CUDA rsqrt(),
+// but it depends on the calling function. We needs to optimize on an individual basis.
+template <class _Tp>
+[[nodiscard]] _CCCL_API _CCCL_FORCEINLINE _Tp __internal_rsqrt_inverse_hyperbloic(_Tp __x) noexcept
+{
+#if _CCCL_CUDA_COMPILATION()
+  if constexpr (is_same_v<_Tp, float>)
+  {
+    NV_IF_TARGET(NV_IS_DEVICE, (return ::rsqrtf(__x);))
+  }
+  if constexpr (is_same_v<_Tp, double>)
+  {
+    NV_IF_TARGET(NV_IS_DEVICE, (return ::rsqrt(__x);))
+  }
+#endif // _CCCL_CUDA_COMPILATION()
+  return _Tp{1} / ::cuda::std::sqrt(__x);
```

Comment:
This could be slightly improved
```suggestion
#if _CCCL_CUDA_COMPILATION()
  if constexpr (is_same_v<_Tp, float>)
  {
    NV_IF_TARGET(NV_IS_DEVICE, (return ::rsqrtf(__x);))
  }
  else if constexpr (is_same_v<_Tp, double>)
  {
    NV_IF_TARGET(NV_IS_DEVICE, (return ::rsqrt(__x);))
  }
  else
#endif // _CCCL_CUDA_COMPILATION()
  {
    return _Tp{1} / ::cuda::std::sqrt(__x);
  }
```